R/xgx_stat_ci.R

```diff
@@ -99,87 +99,150 @@
 #' @importFrom binom binom.exact
 #' @importFrom ggplot2 stat_summary
 #' @importFrom ggplot2 aes
-#' @importFrom ggplot2 position_dodge
+#' @importFrom ggplot2 position_dodeg
 #' @export
-xgx_stat_ci <- function(mapping = NULL, data = NULL, conf_level = 0.95,
-                        distribution = "normal",
-                        geom = list("point", "line", "errorbar"),
-                        position = "identity",
-                        fun.args = list(),
-                        na.rm = FALSE,
-                        show.legend = NA,
-                        inherit.aes = TRUE,
-                        ...) {
-  if (!(conf_level > 0.5 && conf_level < 1)) {
-    stop("conf_level should be greater than 0.5 and less than 1")
-  }
-
-  percentile_value <- conf_level + (1 - conf_level) / 2
-
-  conf_int <- function(y, conf_level, distribution) {
-    y <- stats::na.omit(y)
-
-    if (distribution == "normal") {
-      conf_int_out <- data.frame(
-        y = mean(y),
-        ymin = mean(y) - stats::qt(percentile_value,
-                                   length(y)) * sqrt(stats::var(y) / length(y)),
-        ymax = mean(y) + stats::qt(percentile_value,
-                                   length(y)) * sqrt(stats::var(y) / length(y))
+xgx_stat_ci <- function(
+                    data = NULL,
+                    mapping = NULL,
+                    geom = list("point", "line", "errorbar"),
+                    position = "identity",
+                    na.rm = FALSE,
+                    show.legend = NA, 
+                    inherit.aes = TRUE,
+                    
+                    conf_level = 0.95,
+                    distribution = "normal",
+                    bins = NULL,
```

Comment:
can we add a "breaks" parameter that supplies the cut points for the x bins? That way people can define their own bins if they want to. If breaks is not null, then it would overwrite bins.